### ISSUE TITLE
[a11y]: Tag with filter has aria-label violation

### ISSUE THREAD

**Issue by kglickman** [carbon-design-system/carbon#13706]
### Package

@carbon/react

### Package version

1.23.0

### React version

16.4

### Automated testing tool and ruleset

IBM Equal accessibility checker - latest deployment

### Description

This happens in the Carbon React story book in the playground when setting Filter to true.  On the accessibility tab, it's just a warning.  In the accessibility checker, it's an error.

![image](https://user-images.githubusercontent.com/30597411/235736111-6eda04d4-8bb6-467b-8158-63284b7d257f.png)

### Reproduction/example

Reproducible in story book

### Steps to reproduce

See storybook for Tag.  
Go to playground.  
Set Filter to true.  
Go to A11y tab
Also run Accessibility checker.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
It seems like we can resolve this issue by simply removing the `aria-label` on the `filter` variant. A quick removal via dev tools resolves the error and the voiceover utility seems to still read properly.

**Comment by tay1orjones** [comment]
[Reported on slack](https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1684761247699849), this is also present on v7.x

**Comment by tay1orjones** [comment]
@guidari I'm going to re-open this since we need to backport the fix to v10

**Comment by francinelucca** [comment]
@guidari does this issue need to be closed?

**Comment by guidari** [comment]
> @guidari does this issue need to be closed?

Yes! I don't know why it didn't close after this PR merge: 
https://github.com/carbon-design-system/carbon/pull/13882

I'll close. Thank you for letting me know

**Comment by guidari** [comment]
This issue was resolved in version 11 and 10 by these 2 PRs:

v11: https://github.com/carbon-design-system/carbon/pull/13852
v10: https://github.com/carbon-design-system/carbon/pull/13882
